x-pack/plugins/rule_registry/common/mapping_from_field_map.ts

```diff
@@ -11,7 +11,7 @@ import { FieldMap } from './field_map/types';
 
 export function mappingFromFieldMap(
   fieldMap: FieldMap,
-  dynamic: 'strict' | boolean
+  dynamic: estypes.MappingDynamicMapping
```

Comment:
We would prefer not to turn on dynamic mapping for these alerts as data indices so that we can control which indices are getting mapped. It looks like you're using `dynamic: true` in order to avoid explicitly mapping the `labels` object? Is the `labels` object ECS compliant? If so, can I request we explicitly add the mappings for that object for the APM rule registry indices?

Reply:
the `labels` space is reserved for custom fields in ECS: https://www.elastic.co/guide/en/ecs/8.6/ecs-base.html#field-labels

Reply:
Gotcha. I did not realize. Will the data underneath labels need to be searchable/aggregatable? Is it possible to keep this object defined but unmapped?

Reply:
For this feature, we need to support filtering by `label.*` that users use frequently in the form of KQL.

Reply:
That's fine! The default mapping for the `labels` field in ECS is `type: object` [reference](https://github.com/elastic/ecs/blob/main/generated/elasticsearch/legacy/template.json) I did a quick test and it looks like just using the `object` mapping allows sub-properties to be indexed and searchable.